src/test/java/com/williamfiset/algorithms/datastructures/balancedtree/TreapTreeTest.java

```diff
@@ -0,0 +1,164 @@
+    assertThat(tree.root.left.getValue()).isEqualTo(10);
+    assertThat(tree.root.getValue()).isEqualTo(15);
+    assertThat(tree.root.right.getValue()).isEqualTo(20);
+    assertThat(tree.root.right.right.getValue()).isEqualTo(30);
+
+    tree.insert(32, 14);
+
+    assertThat(tree.root.left.getValue()).isEqualTo(10);
+    assertThat(tree.root.getValue()).isEqualTo(15);
+    assertThat(tree.root.right.getValue()).isEqualTo(32);
+    assertThat(tree.root.right.left.getValue()).isEqualTo(20);
+    assertThat(tree.root.right.left.right.getValue()).isEqualTo(30);
+
+    assertThat(tree.root.right.left.right.left).isNull();
+    assertThat(tree.root.right.left.right.right).isNull();
+    assertThat(tree.root.right.left.left).isNull();
+    assertThat(tree.root.right.right).isNull();
+    assertThat(tree.root.left.left).isNull();
+    assertThat(tree.root.left.right).isNull();
+  }
+
+  @Test
+  public void testLeftRightCase() {
+    tree.insert(20, 10);
+    tree.insert(17, 5);
+    tree.insert(26, 7);
+
+    assertThat(tree.root.getValue()).isEqualTo(20);
+    assertThat(tree.root.left.getValue()).isEqualTo(17);
+    assertThat(tree.root.right.getValue()).isEqualTo(26);
+
+    tree.insert(18, 15);
+    assertThat(tree.root.getValue()).isEqualTo(18);
+    assertThat(tree.root.left.getValue()).isEqualTo(17);
+    assertThat(tree.root.right.getValue()).isEqualTo(20);
+    assertThat(tree.root.right.right.getValue()).isEqualTo(26);
+
+    assertThat(tree.root.left.left).isNull();
+    assertThat(tree.root.left.right).isNull();
+    assertThat(tree.root.right.left).isNull();
+    assertThat(tree.root.right.right.left).isNull();
+    assertThat(tree.root.right.right.right).isNull();
+  }
+
+  @Test
+  public void testRightRightCase() {
+    tree.insert(10, 2);
+    tree.insert(8, 1);
+
+    assertThat(tree.root.getValue()).isEqualTo(10);
+    assertThat(tree.root.left.getValue()).isEqualTo(8);
+
+    tree.insert(7, 3);
+
+    assertThat(tree.root.getValue()).isEqualTo(7);
+    assertThat(tree.root.right.getValue()).isEqualTo(10);
+    assertThat(tree.root.right.left.getValue()).isEqualTo(8);
+
+    assertThat(tree.root.left).isNull();
+    assertThat(tree.root.right.right).isNull();
+    assertThat(tree.root.right.left.right).isNull();
+    assertThat(tree.root.right.left.left).isNull();
+  }
+
+  @Test
+  public void testRightLeftCase() {
+    tree.insert(15, 10);
+    tree.insert(16, 8);
+
+    assertThat(tree.root.getValue()).isEqualTo(15);
+    assertThat(tree.root.right.getValue()).isEqualTo(16);
+
+    tree.insert(13, 11);
+
+    assertThat(tree.root.getValue()).isEqualTo(13);
+    assertThat(tree.root.right.getValue()).isEqualTo(15);
+    assertThat(tree.root.right.right.getValue()).isEqualTo(16);
+
+    assertThat(tree.root.left).isNull();
+    assertThat(tree.root.right.left).isNull();
+    assertThat(tree.root.right.right.left).isNull();
+    assertThat(tree.root.right.right.right).isNull();
+  }
+
+  @Test
+  public void randomRemoveTests() {
```

Comment:
Maybe rename this test to randomTreapOperations since it's doing more than just removing?